src/murmur/Meta.cpp

```diff
@@ -293,10 +293,32 @@ void MetaParams::read(QString fname) {
 		}
 	}
 
-	qsPassword            = typeCheckedFromSettings("serverpassword", qsPassword);
-	usPort                = static_cast< unsigned short >(typeCheckedFromSettings("port", static_cast< uint >(usPort)));
-	iTimeout              = typeCheckedFromSettings("timeout", iTimeout);
-	iMaxTextMessageLength = typeCheckedFromSettings("textmessagelength", iMaxTextMessageLength);
+	QString qsVoiceProtocol = qsSettings->value("voiceProtocolPreference", QString()).toString();
+	if (! qsVoiceProtocol.isEmpty()) {
+		for(const QString &protocol_name: qsHost.split(',')) {
+			QString protocol_upper = protocol_name.trimmed().toUpper();
+			if (protocol_upper.isEmpty()) continue;
+
+			VoiceProtocol protoc(protocol_upper.toStdString());
+			if (protoc.protocolType != VoiceProtocolType::UNSUPPORTED){
+				allowedVoiceProtocolTypes.append(protoc.protocolType);
+				qInfo("Meta: Add %s into voice protocol preference list.", qPrintable(protocol_upper));
+			} else {
+				qWarning("Meta: Ignore unsupported voice protocol %s.", qPrintable(protocol_upper));
```

Comment:
Given that this is read from a server's configuration, I think at this point we should even turn this into a terminating error. Otherwise a server admin could not notice that their settings are not being respected (and this is a rather important one)

Reply:
Great!